### ISSUE TITLE
Custom configurable deploy stages

### ISSUE THREAD

**Issue by distorhead** [flant/werf#1587]
User can specify something like `werf.io/deploy-stage=xxx` for resource to assign manifest update and readiness check of this resource to some stage.

**Comment by ilya-lesikov** [comment]
**TL;DR: Now resources can be deployed in stages. Order of deployment can be configured via `werf.io/weight: "<int>"` annotation, the lesser, the earlier resources with the same weight will be deployed.**

## Deployment stages

Since v1.2.110 (alpha) deployment in stages is available. Annotation `werf.io/weight` can be attached to any resources except Helm-hooks (use `helm.sh/hook-weight` for them). Annotation `werf.io/weight` works just the same as `helm.sh/hook-weight` ([docs](https://helm.sh/docs/topics/charts_hooks/)): resources grouped by their weight and then these groups of resources deployed one after another, starting from the group with the smallest weight, then going for the groups with higher weights. Weight can be 0, a positive or a negative number, which should be passed to annotation as a string. By default, all resources have `werf.io/weight: "0"`.

## Example

This set of resources:

```yaml
kind: Statefulset
metadata:
  name: database
  annotations:
    werf.io/weight: "-10"

kind: Deployment
metadata:
  name: main-app

kind: ConfigMap
metadata:
  name: main-app-cm

kind: Deployment
metadata:
  name: dependent-app
  annotations:
    "werf.io/weight: "10"
```

Will create the following deployment stages:

* Stage 0 (weight -10)
  - database
* Stage 1 (weight 0)
  - main-app
  - main-app-cm
* Stage 2 (weight 10)
  - dependent-app

Every next stage starts only when the resources of the previous stage successfully deployed. Resources of  the same stage deployed concurrently. Error during deployment of any stage stops the whole deployment process.
